### PR TITLE
Action icon as a button mod

### PR DESCRIPTION
**Pull request by jeremie-lucca**
## Description

_ActionIcon_ component as _Button_ mod

-----

Action icon & Button were merged as a single button component during V3 components project on Figma. It seems logical to follow the same logic on Lucca Front as they only have a padding difference. 

- New mod created `.mod-onlyIcon` (open to discussion. `.mod-icon` would be better but already taken by _Icon & Text_ option )
- Action Icon is now deprecated (Deleted with LF19.1)
- Should we keep `mod-delete` (critical background on hover for trash icon)  ?
- `/button/html&css` folder was recfactored to follow last documentation guidelines (every exemples are now visible without toggling options)

```
<button type="button" class="button mod-onlyIcon"><span aria-hidden="true" class="lucca-icon icon-heart"></span></button>
<button type="button" class="button mod-onlyIcon mod-outlined"><span aria-hidden="true" class="lucca-icon icon-heart"></span></button>
<button type="button" class="button mod-onlyIcon mod-text"><span aria-hidden="true" class="lucca-icon icon-heart"></span></button>
<button type="button" class="button mod-onlyIcon mod-outlined"><span aria-hidden="true" class="lucca-icon icon-heart"></span></button>
<button type="button" class="button mod-onlyIcon mod-outlined mod-S"><span aria-hidden="true" class="lucca-icon icon-heart"></span></button>
<button type="button" class="button mod-onlyIcon mod-outlined mod-XS"><span aria-hidden="true" class="lucca-icon icon-heart"></span></button>
```

Capture d’écran 2023-12-01 à 16 08 12

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2438/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/10344/ui-diff)

File: packages/scss/src/components/button/mods.scss [hunk 1]
```diff
@@ -68,6 +72,18 @@
 	}
 }
 
+@mixin onlyIcon {
+	padding: var(--spacings-XS);
```
Comment (vvalentin-lucca) [inline packages/scss/src/components/button/mods.scss:76]: Wouldn't a CSS variable be nice here?

Comment (jeremie-lucca) [reply packages/scss/src/components/button/mods.scss]: Ah, yes, and we already had one. 
Fixed.

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2438/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/10364/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2438/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/11230/ui-diff)

**Review by vvalentin-lucca — DISMISSED** [review]

File: packages/scss/src/components/button/mods.scss [hunk 2]
```diff
@@ -68,19 +72,31 @@
 	}
 }
 
+@mixin onlyIcon {
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/button/mods.scss:75]: we don't add ($atRoot: 'without: rule') here ?

Comment (vvalentin-lucca) [reply packages/scss/src/components/button/mods.scss:76]: There's no sub-component here, so it's not necessary. As a result, I've removed the parameter on mixin `withIcon`.

**Review by AlexiaAlarza — APPROVED** [review]

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2438/storybook

**Review by vvalentin-lucca — APPROVED** [review]

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/11342/ui-diff)

**Review by GuillaumeNury — APPROVED** [review]

**Pull request merged**